src/components/application_manager/src/help_prompt_manager_impl.cc

```diff
@@ -0,0 +1,407 @@
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (vr_commands_.empty()) {
+    LOG4CXX_DEBUG(logger_, "vr_commands_ is empty");
+    return;
+  }
+  LOG4CXX_DEBUG(logger_, "UI request for appID:" << app_.app_id());
+  smart_objects::SmartObjectSPtr ui_global_properties =
+      utils::MakeShared<smart_objects::SmartObject>(
+          smart_objects::SmartType_Map);
+  if (ui_global_properties) {
+    smart_objects::SmartObject& ref = *ui_global_properties;
+
+    ref[strings::params][strings::message_type] =
+        hmi_apis::messageType::request;
+    ref[strings::params][strings::protocol_version] =
+        commands::CommandImpl::protocol_version_;
+    ref[strings::params][strings::protocol_type] =
+        commands::CommandImpl::hmi_protocol_type_;
+    ref[strings::params][strings::correlation_id] =
+        app_manager_.GetNextHMICorrelationID();
+
+    smart_objects::SmartObject& so_to_send = *ui_global_properties;
+    so_to_send[strings::params][strings::function_id] =
+        hmi_apis::FunctionID::UI_SetGlobalProperties;
+
+    smart_objects::SmartObject msg_params =
+        smart_objects::SmartObject(smart_objects::SmartType_Map);
+
+    CreateVRMsg(msg_params);
+
+    msg_params[strings::app_id] = app_.app_id();
+    so_to_send[strings::msg_params] = msg_params;
+    app_manager_.GetRPCService().ManageHMICommand(ui_global_properties);
+  }
+  is_ui_send_ = false;
+}
+
+void HelpPromptManagerImpl::SendBothRequests() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  SendTTSRequest();
+  SendUIRequest();
+}
+
+void HelpPromptManagerImpl::SendRequests() {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  sync_primitives::AutoLock lock(vr_commands_lock_);
+  if (vr_commands_.empty()) {
+    LOG4CXX_DEBUG(logger_, "vr_commands_ is empty");
+    return;
+  }
+
+  switch (sending_type_) {
+    case SendingType::kSendHelpPrompt:
+      SendTTSRequest();
+      return;
+    case SendingType::kSendVRHelp:
+      SendUIRequest();
+      return;
+    case SendingType::kSendBoth:
+      SendBothRequests();
+      return;
+    case SendingType::kNoneSend:
+      break;
+  }
+  LOG4CXX_DEBUG(logger_,
+                "SendingType:" << static_cast<uint32_t>(sending_type_)
+                               << " request not sending");
+}
+
+void HelpPromptManagerImpl::CreatePromptMsg(
+    smart_objects::SmartObject& out_msg_params) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (vr_commands_.empty()) {
+    LOG4CXX_DEBUG(logger_, "vr_commands_ is empty");
+    return;
+  }
+  out_msg_params[strings::help_prompt] =
+      smart_objects::SmartObject(smart_objects::SmartType_Array);
+  uint32_t index = 0;
+  for (const auto& it : vr_commands_) {
+    for (std::size_t i = 0; i < it.second->length(); ++i) {
+      smart_objects::SmartObject item(smart_objects::SmartType_Map);
+
+      item[strings::text] = it.second->getElement(i).asString();
+      item[strings::type] = mobile_apis::SpeechCapabilities::SC_TEXT;
+
+      out_msg_params[strings::help_prompt][index++] = item;
```

Comment:
Same here:
```cpp
if (index == kLimitCommand) {
  app_.set_vr_help(out_msg_params[strings::vr_help]);
  return;
}
```

Reply:
@jacobkeeler this case is covered by `AddCommand` function. It is impossible to add more than 30 commands so this function will never exceed the existing limitations. This check here would be redundant.